components/camera/videosource/helpers.go

```diff
@@ -15,9 +15,27 @@ import (
 	"go.viam.com/rdk/utils"
 )
 
-func decodeImage(imgData []byte) (image.Image, error) {
-	img, _, err := image.Decode(bytes.NewBuffer(imgData))
-	return img, err
+func checkMimeType(ctx context.Context, data []byte) (string, error) {
+	usedMimeType := http.DetectContentType(data)
+	requestedMime := gostream.MIMETypeHint(ctx, "")
+	if requestedMime == utils.MimeTypeRawRGBA {
+		if usedMimeType != utils.MimeTypeDefault {
+			return "", errors.Errorf(
+				"attempted to decode data using %s format as raw rgba data, "+
+					"raw rgba data must encoded with our custom header",
+				usedMimeType,
+			)
+		}
+		formatMatch := bytes.Compare(data[:4], rimage.RGBABitmapMagicNumber)
+		if formatMatch != 0 {
+			return "", errors.New(
+				"attempted to decode raw rgba data, but data was not encoded with the expected header format")
+		}
+	}
+	if requestedMime == "" {
+		requestedMime = usedMimeType
+	}
+	return requestedMime, nil
```

Comment:
I think this is still an issue.

Reply:
sorry I missed this last time, but I have some concerns that I outlined in response to the next comment

Reply:
if I return detectedMimeType here and I use it for the input for image.DecodeImage in readColorURL and readDepthURL, won't it remove the functionality of being able to return a lazy image?

Reply:
Ah, yes, that's correct. If the requestedMime was lazy, then this should returned the requestedMime (after checking that the detectedMimeType is the non-lazy version of the same mime type). Otherwise, it should return the detectedMimeType. This would handle the case where the requestedMime is png and the detectedMimeType is jpeg, for example. In that case, we don't want to pass png to rimage.DecodeImage(). (I'm assuming rimage.DecodeImage() wants the actual mime type rather than the requested mime type? If not, maybe both this function and that one should have comments stating what mime type represents.)

Reply:
Can you respond to this before I take another pass? OR do you want to talk about it in person?